src/pages/_app.tsx

```diff
@@ -6,21 +6,46 @@ import Footer from '@/components/Footer/Footer'
 import dynamic from 'next/dynamic'
 import Web3Network from '@/context/Web3Context'
 import CircleAPIContext from '@/context/CircleAPIContext'
+import { Session } from 'next-auth'
+import { SessionProvider } from 'next-auth/react'
+import { WagmiConfig, createClient, configureChains } from 'wagmi'
+import { avalanche, bsc, mainnet } from '@wagmi/core/chains'
+import { publicProvider } from 'wagmi/providers/public'
 
 const ClientNavbar = dynamic(() => import('../components/Navbar/Navbar'), {
     ssr: false,
 })
 
-export default function App({ Component, pageProps }: AppProps) {
+export const { chains, provider } = configureChains(
+    [mainnet, avalanche],
+    [publicProvider()]
+)
+
+const client = createClient({
+    autoConnect: true,
+    provider,
+})
+
+export default function App({
+    Component,
+    pageProps,
+}: AppProps<{ session: Session }>) {
     return (
         <ChakraProvider theme={theme}>
-            <Web3Network>
-                <CircleAPIContext>
-                    <ClientNavbar />
-                    <Component {...pageProps} />
-                    <Footer />
-                </CircleAPIContext>
-            </Web3Network>
+            <WagmiConfig client={client}>
+                <Web3Network>
+                    <SessionProvider
+                        session={pageProps.session}
+                        refetchInterval={0}
```

Comment:
what does 0 do? is it doest refetch at all?

Reply:
When refetchInterval is set to 0 (the default) there will be no session polling.

If set to any value other than zero, it specifies in seconds how often the client should contact the server to update the session state. If the session state has expired when it is triggered, all open tabs/windows will be updated to reflect this.

Reply:
thanks, should have RTFM but couldnt find the M